src/TestEnvironment.Docker.Containers.Postgres/PostgresContainerWaiter.cs

```diff
@@ -6,39 +6,33 @@
 
 namespace TestEnvironment.Docker.Containers.Postgres
 {
-    public class PostgresContainerWaiter : IContainerWaiter<PostgresContainer>
+    public class PostgresContainerWaiter : BaseContainerWaiter<PostgresContainer>
     {
-        private readonly ILogger _logger;
-
         public PostgresContainerWaiter(ILogger logger)
+            : base(logger)
         {
-            _logger = logger;
         }
 
-        public async Task<bool> Wait(PostgresContainer container, CancellationToken cancellationToken)
+        protected override async Task<bool> PerformCheckAsync(PostgresContainer container,
+            CancellationToken cancellationToken)
         {
-            if (container == null) new ArgumentNullException(nameof(container));
-
             try
             {
-                _logger?.LogInformation($"Postgres: checking container state...");
-                using (var connection = new NpgsqlConnection(container.GetConnectionString()))
-                using (var command = new NpgsqlCommand("select version()", connection))
-                {
-                    await connection.OpenAsync();
-                    await command.ExecuteNonQueryAsync();
-                }
+                Logger?.LogInformation($"Postgres: checking container state...");
```

Comment:
Also we need to unify such information logging in waiters - apply in the same way for all or remove

Reply:
Extracted to `BaseContainerWaiter`